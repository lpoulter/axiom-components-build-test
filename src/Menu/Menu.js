import PropTypes from 'prop-types';
import React, { Component } from 'react';
import omit from 'lodash.omit';
import Base from '../Base/Base';
import './Menu.css';

export default class Menu extends Component {
  static propTypes = {
    /** MenuItems */
    children: PropTypes.node,
    /** Size of the menu */
    size: PropTypes.oneOf(['medium', 'large']),
  };

  static childContextTypes = {
    size: PropTypes.string.isRequired,
  };

  static defaultProps = {
    size: 'large',
  };

  getChildContext() {
    return {
      size: this.props.size,
    };
  }

  render() {
    const { children, ...rest } = this.props;

    return (
      <Base { ...omit(rest, ['size']) } Component="ul" className="ax-menu">
        { children }
      </Base>
    );
  }
}
