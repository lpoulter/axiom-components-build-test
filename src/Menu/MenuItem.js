import PropTypes from 'prop-types';
import React, { Component } from 'react';
import classnames from 'classnames';
import Base from '../Base/Base';

export default class MenuItem extends Component {
  static propTypes = {
    /** Applies styling to indicate it is active */
    active: PropTypes.bool,
    /** Content inside the MenuItem */
    children: PropTypes.node,
    /** Applies styling to indicate it is disabled, and blocks interactions */
    disabled: PropTypes.bool,
  };

  static contextTypes = {
    size: PropTypes.string.isRequired,
  };

  render() {
    const { size } = this.context;
    const { children, active, disabled, ...rest } = this.props;
    const classes = classnames('ax-menu__item', {
      'ax-menu__item--active': active,
      'ax-menu__item--disabled': disabled,
    });

    const textSize = {
      medium: 'headtitle',
      large: 'headline',
    }[size];

    return (
      <Base
          Component="li"
          className={ classes }
          textSize={ textSize }
          textStrong>
        <button { ...rest } className="ax-menu__item-button" disabled={ disabled }>
          { children }
        </button>
      </Base>
    );
  }
}
