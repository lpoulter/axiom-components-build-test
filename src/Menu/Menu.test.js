import React from 'react';
import renderer from 'react-test-renderer';
import Menu from './Menu';
import MenuItem from './MenuItem';

const getComponent = (props = {}) =>
  renderer.create(
    <Menu { ...props }>
      <MenuItem>Lorem ipsum</MenuItem>
      <MenuItem>Lorem ipsum</MenuItem>
    </Menu>
  );

describe('Menu', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size medium', () => {
    const component = getComponent({ size: 'medium' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with size large', () => {
    const component = getComponent({ size: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
