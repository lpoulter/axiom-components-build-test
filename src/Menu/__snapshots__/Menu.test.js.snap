// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders with defaultProps 1`] = `
<ul
  className="ax-menu"
>
  <li
    className="ax-menu__item ax-text--size-headline ax-text--strong"
  >
    <button
      className="ax-menu__item-button"
    >
      Lorem ipsum
    </button>
  </li>
  <li
    className="ax-menu__item ax-text--size-headline ax-text--strong"
  >
    <button
      className="ax-menu__item-button"
    >
      Lorem ipsum
    </button>
  </li>
</ul>
`;

exports[`Menu renders with size large 1`] = `
<ul
  className="ax-menu"
>
  <li
    className="ax-menu__item ax-text--size-headline ax-text--strong"
  >
    <button
      className="ax-menu__item-button"
    >
      Lorem ipsum
    </button>
  </li>
  <li
    className="ax-menu__item ax-text--size-headline ax-text--strong"
  >
    <button
      className="ax-menu__item-button"
    >
      Lorem ipsum
    </button>
  </li>
</ul>
`;

exports[`Menu renders with size medium 1`] = `
<ul
  className="ax-menu"
>
  <li
    className="ax-menu__item ax-text--size-headtitle ax-text--strong"
  >
    <button
      className="ax-menu__item-button"
    >
      Lorem ipsum
    </button>
  </li>
  <li
    className="ax-menu__item ax-text--size-headtitle ax-text--strong"
  >
    <button
      className="ax-menu__item-button"
    >
      Lorem ipsum
    </button>
  </li>
</ul>
`;
